ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -84,6 +84,24 @@ public boolean isEqual(String other) {
     return id.toString().equals(other);
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    AmbryPartition partition = (AmbryPartition) o;
+    return id.equals(partition.id);
```

Comment:
not directly related to this patch, but can you change Long id into long id. No need to use a boxed long if it is non-nullable.

Reply:
Done